### PR TITLE
feat(cirrus): Support non-root user

### LINKED ISSUE #10459: Add non-root user to Cirrus image
In order to run the Cirrus image in a container in the GCP v2 environment, we need to run it as a non-root user due to security policies. Running the image locally, I don't see a non-root user for this purpose. Please add a non-root user to the Cirrus image.

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- Fxa can run the cirrus image in a container in the GCP v2 environment as a non-root user only due to security policies.

This commit

- Support cirrus images to run as non-root users too.

Fixes #10459

### CODE REVIEW

File: cirrus/README.md [hunk 1]
```diff
@@ -47,6 +47,22 @@ To set up the Cirrus environment, follow these steps:
 
 By following these steps, you will create the `.env` file and configure the necessary environment variables for the Cirrus application.
 
+## Running as Non-Root User
+
+By default, the Cirrus Docker image runs the application as the root user. However, if you prefer to run the application as a non-root user for security reasons, you can build the Docker image with additional parameters.
```
Comment (jaredlockhart) [inline cirrus/README.md:52]: Is this true now?  Like if you don't pass in those parameters, it'll use the new 'myuser/1000/1000' user defined below, so will it ever run as root?

File: cirrus/server/Dockerfile [hunk 2]
```diff
@@ -30,6 +30,14 @@ RUN poetry config virtualenvs.create false && \
 # Second stage: Deploy stage
 FROM python:3.11-slim-bullseye as deploy
 
+# Create a non-root user
+ARG USERNAME=myuser
```
Comment (jaredlockhart) [inline cirrus/server/Dockerfile:34]: Let's use `cirrus` for the username?

**Review by jaredlockhart — DISMISSED** [review]
Okay cool so it's a pretty simple change, and thank you for updating the docs as well, just a couple questions 🙏

File: cirrus/server/Dockerfile [hunk 3]
```diff
@@ -46,5 +54,11 @@ ENV PYTHONPATH=$PYTHONPATH:/application-services
 # Copy all other files into the container's working directory
 COPY . .
 
+# Change ownership of the working directory to the non-root user
+RUN chown -R $USERNAME:$USERNAME /cirrus
+
+# Switch to non-root user
+USER $USERNAME
+
```
Comment (jrbenny35) [inline cirrus/server/Dockerfile:62]: Should you switch to the non-root user? This would make this user the default

**Review by jrbenny35 — COMMENTED** [review]
Had a question

File: cirrus/README.md (hunk 1 above)
Comment (yashikakhurana) [reply cirrus/README.md:52]: I have tried it locally to build, by default the root user uses `1000` as a first user id, without any changes in other places,I mean without providing build arguments it worked, can you double check in your local to make sure? I want to make sure it doesn't impact monitor deployment.

File: cirrus/server/Dockerfile (hunk 3 above)
Comment (yashikakhurana) [reply cirrus/server/Dockerfile:62]: sorry I am not sure :(, do you mean why we need to switch?

File: cirrus/README.md (hunk 1 above)
Comment (jaredlockhart) [reply cirrus/README.md:52]: So on my local, on main I get

```
[app-sbAM2aaX-py3.11]jaredlockhart@Jareds-MacBook-Pro~/Documents/experimenter(main)$ docker run -ti cirrus:deploy bash
root@3a1326278208:/cirrus# id
uid=0(root) gid=0(root) groups=0(root)
```

and then in your branch i get

```
[app-sbAM2aaX-py3.11]jaredlockhart@Jareds-MacBook-Pro~/Documents/experimenter(10459/add_non_root_user)$ docker run -ti cirrus:deploy bash
cirrus@92a4a09a84dc:/cirrus$ id
uid=1000(cirrus) gid=1000(cirrus) groups=1000(cirrus)
```

so without any args it's using the new user defined in the args, which is different than the default behaviour now.  

The good news here is this is all taking place inside the container so in theory it shouldn't matter to the external consumer, and using non root inside the container is probably a good security precaution, so it should be okay to just default to always using the new non-root user inside the container, I don't think that should break existing deployments, and even if it does it's just a one time fix.

So it's probably fine to leave this as it is and then just sayin the docs 'by default it runs as cirrus/1000/1000 and if you want to override it to something else just pass these args'.

Comment (yashikakhurana) [reply cirrus/README.md:52]: awesome, thank you for testing that, yeah if we don't provide it uses 1000 default value,  I don't think it should break the existing deployment, but I will update the docs too, and will keep an eye on monitor logs

File: cirrus/server/Dockerfile (hunk 3 above)
Comment (jrbenny35) [reply cirrus/server/Dockerfile:62]: Oh I see, I missed the variable name!

**Pull request merged**